File: math_algorithm/editorial/ruby/prob2-5-4.rb
def is_prime(n)
  return false if n <= 1

  (2..Math.sqrt(n)).each do |i|
    return false if n % i == 0
  end

  true
end

n = gets.chomp.to_i
a = []

(2..n).each do |i|
  a << i if is_prime(i)
end

puts a.join(' ')

File: math_algorithm/editorial/go/prob2-5-4.go
package main

import (
	"fmt"
	"math"
)

func isPrime(n int) bool {
	if n <= 1 {
		return false
	}

	for i := 2; i <= int(math.Sqrt(float64(n))); i++ {
		if n%i == 0 {
			return false
		}
	}

	return true
}

func main() {
	// 入力
	var n int
	fmt.Scanln(&n)
	a := []int{}

	// 素数の検出
	for i := 2; i <= n; i++ {
		if isPrime(i) {
			a = append(a, i)
		}
	}

	// 出力
	for _, num := range a {
		fmt.Printf("%d ", num)
	}
	fmt.Println()
}
